src/componentLoader.js

```diff
@@ -0,0 +1,20 @@
+import domContentLoaded from './domContentLoaded';
+
+/**
+ * Execute the given function according to the `load` parameter.
+ *
+ * @param {function} init The function to execute.
+ * @param {mixed}    load The loading instructions.
+ */
+export default function componentLoader(init, load = domContentLoaded) {
+  if (init != null) {
+    if (typeof load === 'function') {
+      load(init);
+    } else if (Array.isArray(load)) {
```

Comment:
> I think some of these types could be handled easily with Typescript

Can you expand on this a bit? I hear folks say things like this and I don't understand. We would still need to check the type so that we know which condition passes.

Reply:
Yes, and my statement is a bit simplistic I admit. We still would need to check the type as you mentioned, but we can ensure our types are not `any` type, but instead of a certain type. Let me explain with an example:

```ts
import domContentLoaded from './domContentLoaded';

// Declare the Load parameter type.
type Load = [HTMLElement | Window, string] | Function | boolean;

/**
 * Execute the given function according to the `load` parameter.
 *
 * @param init The function to execute.
 * @param load The loading instructions.
 */
export default function componentLoader(
  init: () => {},
  load: Load = domContentLoaded
) {
  if (init) {
    if (typeof load === 'function') {
      load(init);
    } else if (Array.isArray(load)) {
      // Here we know that the only allowed array is
      // an HTMLElement or Window and a string.
      const [element, event] = load;

      // e.g., load: [window, 'load']
      // We can use optional chaining just to ensure we don't get TypeError.
      element?.addEventListener(event, init);
    }
  }
}

```